File: 53_maximum_subarray/maxSubArray.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        curr_sum = 0
        
        for num in nums:
            if curr_sum < 0:
                curr_sum = 0
            curr_sum += num
            max_sum = max(max_sum, curr_sum)
        
        return max_sum

File: 53_maximum_subarray/maxSubArray.ts
function maxSubArray(nums: number[]): number {
  let max = nums[0];
  let sum = 0;

  for (let num of nums) {
    if (sum < 0) {
      sum = 0;
    }
    sum += num;
    max = Math.max(max, sum);
  }

  return max;
}
